practice-app/website/api/copyright.py

```diff
@@ -0,0 +1,187 @@
+from flask_jwt_extended import current_user, verify_jwt_in_request
+from datetime import date
+
+from .. import db
+import requests
+
+from .jwt import user_required
+
+from ..models import User, ArtItem, CopyrightInfringementReport
+
+copyright = Blueprint("copyright", __name__)
+
+# API endpoint for reporting a copyright infringement
+@copyright.route("/copyright", methods=["POST"])
+@user_required()
+def report_infringement():
+
+    """
+    file: ./doc/copyright_POST.yml
+    """
+
+    # Checking if the request body is JSON
+    if (not request.is_json):
+        return {"error": "Request body should be a JSON file."}, 400
+    
+    request_json = request.get_json()
+    
+    # Checking if the request JSON has all the required fields 
+    if (is_missing_field(request_json, {"original_art_item_id", "infringement_art_item_id", "description"})):
+        return {"error": "You have not provided some of the required fields."}, 400
+
+    # Quering for given art items URIs
+    original_art_item = ArtItem.query.get(request_json["original_art_item_id"])
+    infringement_art_item = ArtItem.query.get(request_json["infringement_art_item_id"])
+
+    if(not original_art_item):
+        return {"error": f"Art Item with id : {request_json['original_art_item_id']} does not exists."}, 404
+    if(not infringement_art_item):
+        return {"error": f"Art Item with id : {request_json['infringement_art_item_id']} does not exists."}, 404
+
+    # Checking if the given art items are different
+    if (request_json["original_art_item_id"] == request_json["infringement_art_item_id"]):
+        return {"error": "Original art item and infringement art item must be different."}, 400
+
+    # Getting a similarity score by calling an external API to compare two images
+    #   Smaller the value, similar the images
+    sim_score = get_similarity_score(original_art_item.content_uri, infringement_art_item.content_uri)
+
+    if (sim_score == -1):
+        return {"error": "Something went wrong in the external Image Similarity API. Please try again."}, 400
+
+    verify_jwt_in_request()
+
+    new_copyright_report = CopyrightInfringementReport(
+        creator = current_user.id,
+        original_art_item_id = request_json["original_art_item_id"],
+        infringement_art_item_id = request_json["infringement_art_item_id"],
+        description = request_json["description"],
+        similarity_score = sim_score,
+        creation_date = date.today()
+    )
+
+    # Creating the report and commiting it 
+    try:
+        db.session.add(new_copyright_report)
+        db.session.commit()
+    except Exception as error:
+        return {"error" : f"There was an error adding the copyright infringement report to the database, Error: {error}."}, 500
+
+    # Success
+    return {"id": new_copyright_report.id}, 201
+
+# API endpoint for viewing a copyright infringement report
+@copyright.route("/copyright/<report_id>", methods=["GET"])
```

Comment:
The data of the copyright report should not be visible to outside imo. This should at least require a user access token.

Reply:
Same as above conversation